cost/azure/old_snapshots/azure_delete_old_snapshots.pt

```diff
@@ -156,26 +288,104 @@ script "js_filter_snapshots", type: "javascript" do
+      values[0] = values[0].substr(0, values[0].length - 3)
+      result = separator + chunk + result
+    }
+    if (values[0].length > 0){
+      result = values[0] + result
+    }
+    if(values[1]==undefined){
+      return result;
+    }
+    return result+"."+values[1];
+  }
+  // Format costs with currency symbol and thousands separator
+  if(ds_billing_centers.length!=0){
+    if( ds_currency_code['value'] !== undefined ) {
+      if (ds_currency_reference[ds_currency_code['value']] !== undefined ) {
+        var cur = ds_currency_reference[ds_currency_code['value']]['symbol']
+        if( ds_currency_reference[ds_currency_code['value']]['t_separator'] !== undefined ) {
+          var separator = ds_currency_reference[ds_currency_code['value']]['t_separator']
+        } else {
+          var separator = ""
+        }
+      } else {
+        var cur = ""
+        var separator = ""
+      }
+    } else {
+      var cur = "$"
+      var separator = ","
+    }
+    var total=0;
+    _.each(snapshots, function(snapshot){
+      var cost_objects = _.where(snapshots_costs, {resource_id: snapshot["snapshot_id"]});
+      if (_.size(cost_objects) > 0){
+        count++;
+        var sum = _.reduce(_.compact(_.map(cost_objects, function(value){ return value.cost_nonamortized_unblended_adj})), function(memo, num){ return memo + num; }, 0);
+        var monthly_savings = sum*30;
+        total=total+monthly_savings;
+        snapshot['savings']= cur + ' '+formatNumber((Math.round(monthly_savings * 100) / 100), separator)
+        instances.push(snapshot)
+      } else {
+        snapshot['savings']= "N/A";
+        instances.push(snapshot)
+      }
+    })
+    if(count){
+      total=cur + ' '+formatNumber((Math.round(total * 100) / 100), separator);
+      message="The total estimated monthly savings are " +total;
+    }
+    else{
+        message="The Flexera Optima system does not have any data to calculate savings for these resources";
+    }
+    }else{
+    message="You do not have the minimum required role of billing_center_viewer to view the savings";
+    instances=snapshots;
+    }
+    result={"instances": instances,
+    "message": message
+   };
+  result.instances = _.sortBy(result.instances,"location");
+  result.instances = _.sortBy(result.instances,"subscriptionName");
   EOS
 end
 
 policy "pol_azure_delete_old_snapshots" do
-  validate $ds_filter_snapshots do
-    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Old Snapshots Found"
+  validate $ds_snapshot_cost_mapping do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data.instances }} Old Snapshots Found"
+  detail_template <<-EOS
+  {{data.message}}.\n
+  EOS
     # Send email report
     escalate $send_email_report
     # Delete snapshot if user approves
     escalate $process_snapshots
-    check eq(size(data), 0)
-    export do
+    check eq(size(val(data, "instances")), 0)
```

Comment:
You may want to add a `hash_include "instances"` as mentioned in [Describing the Policy Conditions](https://docs.rightscale.com/policies/developers/reference/v20180301/policy_template_language.html#describing-the-policy-conditions) to ensure that new notifications only show up when the `instances` field of the `data` changes if there are other items in `data` which may changes.

Reply:
Thanks for the suggestion!
Added hash_include "instances", currently I am testing this change, and it's in progress, I will puh this change to GitHub once it's working fine as expected!

Reply:
Tested hash_include changes, its working as expected, hence pushed this change to GitHub.  Pls review it.